server/src/main/java/com/ncc/neon/better/ExperimentConfig.java

```diff
@@ -0,0 +1,113 @@
+package com.ncc.neon.better;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.ncc.neon.models.ExperimentForm;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+import static com.ncc.neon.controllers.BetterController.SHARE_PATH;
+
+public class ExperimentConfig {
+    public String module;
+
+    private String name;
+    private String trainFile;
+    private String devFile;
+    private String testFile;
+    private Map<String, List<String>> rawConfig;
+    private ArrayList<EvalConfig> evalConfigs;
+
+    public ExperimentConfig(ExperimentForm experimentForm) throws IOException {
+        this.module = experimentForm.getModule();
+        trainFile = experimentForm.getTrainFile();
+        devFile = experimentForm.getDevFile();
+        testFile = experimentForm.getTestFile();
+
+        // Read config file.
+        ObjectMapper objectMapper = new ObjectMapper();
+        File configFile = new File(Paths.get(SHARE_PATH.toString(), experimentForm.getConfigFile()).toString());
+
+        rawConfig = objectMapper.readValue(configFile, Map.class);
+        evalConfigs = new ArrayList<>();
+
+        initName();
+
+        if (rawConfig.size() == 0) {
+            // Add a single config with only required parameters.
+            evalConfigs.add(EvalConfig.buildConfig(module, trainFile, devFile, testFile, name, new ArrayList<>()));
+        }
+        else {
+            parseConfig(experimentForm.getTrainFile(), experimentForm.getDevFile(), experimentForm.getTestFile());
+        }
+    }
+
+    public EvalConfig[] getEvalConfigs() {
+        return evalConfigs.toArray(new EvalConfig[0]);
+    }
+
+    public String getTrainFile() {
+        return trainFile;
+    }
+
+    public String getDevFile() {
+        return devFile;
+    }
+
+    public String getTestFile() {
+        return testFile;
+    }
+
+    public String getName() { return name; }
+
+    private void initName() {
+        TimeZone tz = TimeZone.getTimeZone("UTC");
+        DateFormat df = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm'Z'");

```

Comment:
You can utilize DateUtil.getCurrentDateTime() static method to do this.

Reply:
Good catch.  I moved this init to the constructor and removed the private function.